pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -481,6 +490,11 @@ def reconciliate_processes(self, trace: str):
         if self.world_size < 2:
             return
 
+        # If the cluster environment creates the process, allow the scheduler / parent process
+        # to perform the process termination
+        if not self._has_called_call_children_scripts:
+            return
+
```

Comment:
note to reviewer: this is redundant with the changes made above to `_share_information_to_prevent_deadlock` as `self._sync_dir` is not initialized, which would lead to the early return. however, i include it here to be explicit and defensive in case anything changes with `_share_information_to_prevent_deadlock`

Reply:
Hey @ananthsub,

1. `LOCAL_RANK` not being available,  will result to LightningEnvironment.creates_children() -> False and therefore rank 0 will create the processes.

2. os.environ is populated with LOCAL_RANK, and rank 0 contains `_call_children_scripts=False` and rank > 0 contains `_call_children_scripts=True`

Yes, the only way is to broadcast from rank 0 to the other processes.